js/src/main/kotlin/org/rust/js/RsWasmBindgenLineMarkerProvider.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.js
+
+import com.intellij.codeInsight.daemon.DefaultGutterIconNavigationHandler
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerInfo
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider
+import com.intellij.lang.javascript.psi.JSElement
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptClass
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptEnum
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptFunction
+import com.intellij.navigation.GotoRelatedItem
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.findDescendantOfType
+import icons.JavaScriptPsiIcons
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.RsOuterAttributeOwner
+import org.rust.lang.core.psi.ext.RsVisibilityOwner
+import org.rust.lang.core.psi.ext.containingCargoPackage
+import org.rust.lang.core.psi.ext.findOuterAttr
+import org.rust.lang.core.psi.impl.RsBaseTypeImpl
+import java.nio.file.InvalidPathException
+import java.nio.file.Paths
+
+class RsWasmBindgenLineMarkerProvider : RelatedItemLineMarkerProvider() {
+    override fun getLineMarkerInfo(element: PsiElement): RelatedItemLineMarkerInfo<*>? {
+        val element = element as? RsOuterAttributeOwner ?: return null
+        if (element !is RsFunction && element !is RsEnumItem && element !is RsStructItem && element !is RsImplItem) {
+            return null
+        }
+
+        // Check the package has wasm-bindgen as dependency
+        val cargoPackage = element.containingCargoPackage ?: return null
+        if (cargoPackage.dependencies.none { it.name == "wasm-bindgen" }) {
```

Comment:
Unfortunately, this does not work because `dependency.name` refers to the name inside `Cargo.toml`, potentially aliased. Consider such a case:
```
[dependencies]
my_wasm_bindgen_alias = { package = "wasm-bindgen", version = "0.2.63" }
```
In this case, `dependency.name` is `"my_wasm_bindgen_alias"`, but `dependency.pkg.name` is `"wasm-bindgen"`. So looks like you should use `pkg.name` here